contracts/interfaces/IRegistry.sol

```diff
@@ -2,13 +2,58 @@ pragma solidity ^0.4.23;
 
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20Basic.sol";
+import "./IPausable.sol";
 
-contract IRegistry is Ownable {
+contract IRegistry is Ownable, IPausable {
+
+    /*
+    Valid Address Keys
+        TR_Address -> getAddress("TR_Address")
+        STR_Address -> getAddress("STR_Address")
+        MR_Address -> getAddress("MR_Address")
+        POLY_Address -> getAddress("POLY_Address")
+        */
 
     // Registration fee in POLY base 18 decimals
     uint256 public registrationFee;
 
+    mapping (bytes32 => address) public storedAddresses;
+    mapping (bytes32 => bool) public validAddressKeys;
+
     event LogChangePolyRegisterationFee(uint256 _oldFee, uint256 _newFee);
+    event LogChangeAddress(string _nameKey, address indexed _oldAddress, address indexed _newAddress);
+
+    /**
+     * @dev get the contract address
+     * @param _nameKey is the key for the contract address mapping
+     */
+    function getAddress(string _nameKey) public returns(address) {
+        require(validAddressKeys[keccak256(_nameKey)]);
+        return storedAddresses[keccak256(_nameKey)];
+    }
+
+    /**
+     * @dev set the contract address
+     * @param _nameKey is the key for the contract address mapping
+     * @param _newAddress is the new contract address
+     */
+    function setAddress(string _nameKey, address _newAddress) public onlyOwner {
+        require(!validAddressKeys[keccak256(_nameKey)] && storedAddresses[keccak256(_nameKey)] == address(0));
+        validAddressKeys[keccak256(_nameKey)] = true;
+        storedAddresses[keccak256(_nameKey)] = _newAddress;
+        emit LogChangeAddress(_nameKey, address(0), _newAddress);
+    }
+
+    /**
+     * @dev change the contract address
+     * @param _nameKey is the key for the contract address mapping
+     * @param _newAddress is the new contract address
+     */
+    function changeAddress(string _nameKey, address _newAddress) public onlyOwner {
+        address oldAddress = getAddress(_nameKey);
+        storedAddresses[keccak256(_nameKey)] = _newAddress;
+        emit LogChangeAddress(_nameKey, oldAddress, _newAddress);
+    }
```

Comment:
Can we merge setAddress & changeAddress (so we just have a single changeAddress function which is used both for initialisation and subsequent changes).